EngineLayer/Calibration/CalibrationEngine.cs

```diff
@@ -1,332 +1,252 @@
 ﻿using MassSpectrometry;
-using SharpLearning.Common.Interfaces;
-using SharpLearning.Containers.Matrices;
-using SharpLearning.CrossValidation.TrainingTestSplitters;
-using SharpLearning.GradientBoost.Learners;
-using SharpLearning.GradientBoost.Models;
-using SharpLearning.Metrics.Regression;
-using SharpLearning.Optimization;
-using SharpLearning.RandomForest.Learners;
-using SharpLearning.RandomForest.Models;
 using System;
-using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Linq;
-using System.Threading.Tasks;
 
 namespace EngineLayer.Calibration
 {
     public class CalibrationEngine : MetaMorpheusEngine
     {
-        private const double MaximumFracForTraining = 0.70;
-        private const double MaximumDatapointsToTrainWith = 20000;
-        private const int TrainingIterations = 30;
-        private readonly int RandomSeed;
-
+        private const int NumberOfScansUsedForSmoothingOnEachSide = 100;
         private readonly MsDataFile MyMsDataFile;
         private readonly DataPointAquisitionResults Datapoints;
+        public MsDataFile CalibratedDataFile { get; private set; }
```

Comment:
RAM issues maybe? does doubling-up on storing the spectra causes 2x the RAM usage? or did we do this before?

Reply:
I agree that this would take up double the RAM, but the RAM of an MS file isn't massive (the most I've seen is 3GB). We can get around this by modifying the existing spectra (as was done previously), but that is a little ugly and opens the door for continuous integration issues. I am leaning that we make a new calibrated spectrum instead of modifying an old one.

Reply:
I'm willing to go back to modifying the spectra instead of creating new spectra, but then this will have to wait until mzLib gets updated in MetaMorpheus.